docs/setup-guide/build_k8s_from_source/README.md

```diff
@@ -0,0 +1,145 @@
+# Introduction
+This guide introduces setting up a <b>development environment</b> for Kubernetes (K8s) cluster on a single instance (vm). `kubeadm` is installed for deploying and managing K8s clusters. This setup has been tested on a 2-node cluster.
+
+# Launch a vm on AWS
+- Click 'Launch Instances' on AWS
+- Search 'ubuntu' and select 'Ubuntu Server 18.04 LTE (HVM), SSD Volume Type' with '64-bit (x86)'.
+- Choose the Instance Type 't2.large', and click 'Next...'
+- Change 'Auto-assign Public IP' to 'Enable' and click 'Next...'
+- Chage storage size (GiB) to '100' and click 'Next...'
+- Click 'Add Tag'; Type 'Name' for Key, and '\<vm-tag-name\>' for Value, check 'Instances' only. Click 'Next...'
+- Click 'Add rule' for Inbound rules, following the below list
+```
+All TCP             TCP     0-65535     0.0.0.0/0
+All UDP             UDP     0-65535     0.0.0.0/0
+SSH                 TCP     22          0.0.0.0/0
+All ICMP - IPv4     ICMP    All         0.0.0.0/0
+```
+- Click 'Review and Launch'
+- Click 'Launch'
+- Select a key pair
+- Wait until the vm is running
+
+# Install K8s components and tools
+- SSH into the vm on AWS
+
+## Configure iptables
+- Run `sudo modprobe br_netfilter`
+- Run `lsmod | grep br_netfilter` to make sure br_netfilter module is loaded
+- Run
+  ```
+  cat <<EOF | sudo tee /etc/modules-load.d/k8s.conf
+  br_netfilter
+  EOF
+  ```
+- Run
+  ```
+  cat <<EOF | sudo tee /etc/sysctl.d/k8s.conf
+  net.bridge.bridge-nf-call-ip6tables = 1
+  net.bridge.bridge-nf-call-iptables = 1
+  EOF
+  ```
+- Run `sudo sysctl --system`
+
+## Install runtime docker
+- Run `sudo apt-get -y update; sudo apt-get -y install docker.io`
+- Run `sudo systemctl enable docker.service`
+- Run `sudo groupadd docker`
+- Run `sudo usermod -aG docker $USER`
+- Reboot the machine by running `sudo reboot`
+- Verify you can run `docker` without `sudo` by running `docker run hello-world`
+- Run `sudo mkdir /etc/docker`
+- Run 
+```
+cat <<EOF | sudo tee /etc/docker/daemon.json
+{
+  "exec-opts": ["native.cgroupdriver=systemd"],
+  "log-driver": "json-file",
+  "log-opts": {
+    "max-size": "100m"
+  },
+  "storage-driver": "overlay2"
+}
+EOF
+```
+- Run `sudo systemctl daemon-reload`
+- Run `sudo systemctl restart docker`
+
+## Install GoLang
+- Run `cd ~`
+- Run `sudo apt-get -y install make gcc jq`
+- Run `wget https://redirector.gvt1.com/edgedl/go/go1.17.1.linux-amd64.tar.gz -O go1.17.1.tar.gz`
+- Run `tar -xzvf go1.17.1.tar.gz`
```

Comment:
rm -rf /usr/local/go && tar -C /usr/local -xzf go1.17.1.linux-amd64.tar.gz

Reply:
added `rm -rf /usr/local/go` before installing new go
we still extract the package under $HOME since later on we need to set the GOROOT env var and add soft links.